core/dbt/contracts/graph/manifest.py

```diff
@@ -1550,7 +1550,7 @@ def __init__(self, macros) -> None:
 
 
 @dataclass
-@schema_version("manifest", 11)
+@schema_version("manifest", 12)
```

Comment:
Why bump the manifest version but not the run_results version?

Reply:
I couldn't find documentation on when the different schema versions should be bumped. I knew that we always bumped the manifest version on minor release (and that we yet haven't done so in main since cutting 1.7 into it's release branch). I didn't know when `run-results`, `catalog`, and `sources` should be bumped, and didn't find documentation with that information

Reply:
Don't we want to fix the v11 json schema?  Otherwise we're leaving a broken schema out there. This will have to be backported to 1.7. I guess we could fix the v11 schema in 1.7.latest, but it seems like it would be simpler to just fix the v11 schema and then bump to v12 when there are actual schema changes. (Have there been schema changes since v11 was produced? If there have been, we should generate v11 schema from older class versions.)

Reply:
In my mind v11 manifest should be generated from the objects that exist on the 1.7.latest branch during the backport. Doing so guarantees that the generated manifest jsonschema matches the `WriteableManifest` class. I went through the commit history of `/core/dbt/contracts` of `main` and `1.7.latest`, and they do differ. Looking at the commit differences I don't think there should be a difference. However, the risk we run in regenerating the v11 manifest here is the v11 manifest jsonschema doesn't match the actual written artifacts. I believe partial parsing would then break as we won't be able to instantiate the `WritableManifest` for the `PreviousState`. Additionally, I believe in such a scenario dbt cloud jobs would begin failing because the artifacts will likely fail to pass jsonschema validation. It's fairly easy to reduce this risk to _zero_ by simply regenerating the v11 manifest in the backport itself. Given that it's fairly easy to reduce the risk to _zero_, I recommend we do so. This will be going out just before the holidays, and it would suck if we had to do another fix for this / have an incident.

That said, I'm happy to not generate the v12 manifest in this PR (and also not regenerate the v11 manifest). However that just makes it harder to prove that this PR to main is doing what is supposed to. If that's acceptable though, happy to drop those changes